src/helpers/programHelper.js

```diff
@@ -800,6 +802,78 @@ class ProgramServiceHelper {
 
     return from(axios(req));
   }
+
+  /**
+   * Update the user profile with medium, subject and gradeLevel
+   *
+   * @param integer program_id  Program id
+   * @param integer user_id     User id
+   */
+  onAfterAddNomination(program_id, user_id) {
+    this.getProgramDetails(program_id).then(data => {
+      const value = {};
+      value['body'] = {
+        "id": "open-saber.registry.search",
+        "request": {
+            "entityType":["User"],
+            "filters": {
+              "userId": {
+                  "contains": user_id
+              }
+          }
+        }
+      };
+
+      const callback = (err, res) => {
+        if (!err && res) {
+          const user = _.first(res.data.result.User);
+          const updateRequestBody = {};
+          updateRequestBody['body'] = {
+            "id": "open-saber.registry.update",
+            "ver": "1.0",
+            "request": {
+              "User": {
+               "osid": user.osid,
+               "medium": _.union(user.medium, data.config.medium),
+               "gradeLevel": _.union(user.gradeLevel, data.config.gradeLevel),
+               "subject": _.union(user.subject, data.config.subject)
+               }
+            }
+          };
+          registryService.updateRecord(updateRequestBody, (error, response) => {
+            if (!error && response) {
+              return response;
+            } else {
+              return error;
+            }
+          });
+        }
+      };
+      registryService.searchRecord(value, callback);
```

Comment:
Do we have to perform this search to get user `osid`? Are we not getting it in the frontend so that we can pass that as part of request? But then we will have to change the request of `add nomination` I guess. Just a thought. May be @pallakartheekreddy  can comment?

Reply:
@venkateshwarans We can update user using `userId` instead `osid`. Can you ask amol to fix this

Reply:
@pallakartheekreddy I guess there is no support yet from the platform to update by userId.
 
@venkanagouda-k  Can we pass the osid with nomination add request from frontend?